### PR TITLE
Fix and improve uefistored tests

### PR DESCRIPTION
**Pull request by stormi**
Many fixes and improvements to Guest secure boot tests.

### CODE REVIEW

**Review by Wescoeur — APPROVED** [review]

File: tests/uefistored/conftest.py [hunk 1]
```diff
@@ -1,31 +1,41 @@
...
 @pytest.fixture(scope='module')
-def imported_sb_vm(imported_vm):
-    vm = imported_vm
+def pool_without_uefi_certs(host):
+    pool = host.pool
 
-    if not vm.is_uefi:
-        pytest.skip('imported_sb_vm can only be used on UEFI VMs')
+    # Save the certs.
+    # This may fail if the pool certs are not in a consistent state, and prevent the tests from running.
```
Comment (beshleman) [inline tests/uefistored/conftest.py:9]: I wonder if we should place this in the README too. A gentle reminder about using secureboot-certs to initialize a pool for tests, perhaps?

File: lib/common.py [hunk 2]
```diff
@@ -871,6 +964,95 @@ def clone(self):
...
+                    10
+                )
+                ret = True
+            except TimeOutException as e:
+                logging.debug(e)
+                pass
+        finally:
+            self.host.ssh(['screen', '-S', session, '-X', 'quit'], check=False)
+            self.host.ssh(['rm', '-f', tmp_file], check=False)
+        return ret
```
Comment (beshleman) [inline lib/common.py:1053]: This whole section is not as bad as I was expecting given the comments! :+1:

File: tests/uefistored/conftest.py (hunk 1 above)
Comment (stormi) [reply tests/uefistored/conftest.py:9]: You don't need to initialize the pool with secureboot-certs. The recorded state will just be that no certs are installed, so all certs will be deleted during teardown in this situation.

I've just added a Requirements comment on top of the file that says that the state must be consistent. As there is no way to enter an inconsistent state using secureboot-certs (in theory), I don't think it needs to be in the main README.

File: lib/common.py [hunk 3]
```diff
@@ -111,6 +117,93 @@ def first_shared_sr(self):
...
+                         % (' '.join(saved_certs.keys()) if saved_certs else 'no certs'))
+        else:
+            for tmp_file in saved_certs.values():
+                self.master.ssh(['rm', '-f', tmp_file])
+            raise Exception(
+                (
+                    "Can't save pool UEFI certs. Only %s certs are defined, "
+                    "which wouldn't be restorable as is with secureboot-certs install"
+                )
+                % '&'.join(saved_certs.keys())
```
[the remark this answers is not in the document]
Comment (stormi) [reply lib/common.py]: Done

File: lib/common.py [hunk 4]
```diff
@@ -25,6 +25,9 @@ def vm_image(vm_key):
         url = DEF_VM_URL + url
     return url
 
+class TimeOutException(Exception):
```
[the remark this answers is not in the document]
Comment (stormi) [reply lib/common.py]: Thanks, this is better indeed.

**Pull request merged**